workflow-templates/dreact-ci.yml

```diff
@@ -0,0 +1,62 @@
+name: DReact CI
+on:
+  pull-request:
+    types: [closed]
+
+jobs:
+  main:
+    name: main
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged == true
+    env: # Update for each app
+      APP_SLUG: app-slug
+      APP_TITLE: App Title
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v2
+        with:
+          path: main
+      - name: Checkout Module Repo
+        uses: actions/checkout@v2
+        with:
+          repository: "BrownUniversity/dreact"
+          path: dreact
+          token: ${{ secrets.API_TOKEN_GITHUB }} # PAT
+      - uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - name: Install App Dependencies
+        run: npm ci
+        working-directory: main
+      - name: Build App
+        run: npm run build
+        working-directory: main
+      - name: Copy App to Module
+        run: cp main/dist/${{ env.APP_SLUG }}.js dreact/apps/${{ env.APP_SLUG }}.js
+      - name: Create Pull Request in Module Repo
+        id: pull-request
+        uses: brownuniversity/create-pull-request@v1.0.15
+        with:
+          token: ${{ secrets.API_TOKEN_GITHUB }} # PAT
+          branch-name: feature/${{ env.APP_SLUG }}
+          commit-message: "Update ${{ env.APP_TITLE }} app"
+          title: "Update ${{ env.APP_TITLE }} app"
+          description: "Changes from ${{ github.event.pull_request.html_url }}"
+          draft: true
+          directory: "dreact"
+      - name: Comment on App Pull Request
+        uses: actions/github-script@v3
+        if: env.pr_url
```

Comment:
Did setting an output from the custom action not work? I feel like it's not very clear where this comes from. I need to go look at the readme to learn this, where as `steps.pull-request.outputs.pr_url` would be clear in my opinion.

Reply:
It wasn't working for me, but maybe I messed up the quotes or something.  I'll give it another shot

Reply:
I was just looking through the history. I suspect that things added to $GITHUB_ENV are not actually available in the `env` context until a subsequent step. So maybe you just need to separate setting the output into a followup step?

Reply:
That does [seem to be the case](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable):

> The action that creates or updates the environment variable does not have access to the new value, but all subsequent actions in a job will have access.

Reply:
We may not even need the environment variable? Can we just use the result directly in `set-output` [like this](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#outputs-for-composite-run-steps-actions)?

Reply:
I tried the subshell approach [here](https://github.com/BrownUniversity/create-pull-request/blob/main/action.yml#L53), but it doesn't look like it's working: https://github.com/BrownUniversity/drupal-react-juicer/pull/56

Any ideas?

Reply:
Okay now it looks like the output is getting set but to the text of the bash command instead of having it actually execute: https://github.com/BrownUniversity/drupal-react-juicer/runs/3086950123?check_suite_focus=true

Reply:
bash command substitution doesn't work in single quotes. I think `echo "::set-output name=pr-url::$(gh pr create --title "${{ inputs.title }}" --body "${{ inputs.description }}" ${{ steps.set-draft.outputs.draft }})"` should be fine. Each substituted command (`$(...)`) has its own parsing step.

Reply:
Success! 🎉  Will update the template now